src/renderer/components/layout/main-layout.tsx

```diff
@@ -18,43 +19,74 @@ interface Props {
 
 @observer
 export class MainLayout extends React.Component<Props> {
-  public storage = createStorage("main_layout", { pinnedSidebar: true });
+  public storage = createStorage("main_layout", {
+    pinnedSidebar: true,
+    sidebarWidth: 200,
+  });
 
   @observable isPinned = this.storage.get().pinnedSidebar;
   @observable isAccessible = true;
+  @observable sidebarWidth = this.storage.get().sidebarWidth
 
   @disposeOnUnmount syncPinnedStateWithStorage = reaction(
     () => this.isPinned,
     (isPinned) => this.storage.merge({ pinnedSidebar: isPinned })
   );
 
+  @disposeOnUnmount syncWidthStateWithStorage = reaction(
+    () => this.sidebarWidth,
+    (sidebarWidth) => this.storage.merge({ sidebarWidth })
+  );
+
   toggleSidebar = () => {
     this.isPinned = !this.isPinned;
     this.isAccessible = false;
     setTimeout(() => (this.isAccessible = true), 250);
   };
 
+  getSidebarSize = () => {
+    return {
+      "--sidebar-width": `${this.sidebarWidth}px`,
+    }
+  }
+
+  @autobind()
+  adjustWidth(newWidth: number): void {
+    this.sidebarWidth = newWidth
+  }
+
   render() {
     const { className, headerClass, footer, footerClass, children } = this.props;
     const cluster = getHostedCluster();
     if (!cluster) {
       return null; // fix: skip render when removing active (visible) cluster
     }
     return (
-      <div className={cssNames("MainLayout", className)}>
+      <div className={cssNames("MainLayout", className)} style={this.getSidebarSize() as any}>
         <header className={cssNames("flex gaps align-center", headerClass)}>
           <span className="cluster">{cluster.preferences.clusterName || cluster.contextName}</span>
         </header>
 
         <aside className={cssNames("flex column", { pinned: this.isPinned, accessible: this.isAccessible })}>
           <Sidebar className="box grow" isPinned={this.isPinned} toggle={this.toggleSidebar} />
+          <ResizingAnchor
+            direction={ResizeDirection.HORIZONTAL}
+            placement={ResizeSide.TRAILING}
+            growthDirection={ResizeGrowthDirection.LEFT_TO_RIGHT}
+            getCurrentExtent={() => this.sidebarWidth}
+            onDrag={this.adjustWidth}
+            onDoubleClick={this.toggleSidebar}
```

Comment:
Do we need this? I noticed that when you double click while on the right edge of the sidebar (and don't move the mouse), the sidebar changes to the default size and only hides when you subsequently move the mouse. Probably the toggle is enough?

Also, when pinned, when you hover over the sidebar it grows out to only the default width, not the user-adjusted width. Is that intended?

Reply:
Re second point, do you mean when it is not pinned?

Reply:
Also, probably would be nice if you could resize it when it's pinned too (once the user has hovered over it to open it)

Reply:
yes, sorry, I guess I do. I find pinned to be an ambiguous term here.

Reply:
`isPinned === true` => "still visible when mouse is not hovering"

Reply:
Allowing adjustment while hovering I think is out of scope for this issue. Though we should ask @jakolehm or @aleksfront. The reason being is that the current hovering mechanic is purely css based and it would be necessary to translate this into a JS implementation for dragging while in "non-pinned mode".

Reply:
Also re it re opening after closing. The same happens when using the toggle button that has to do with how `:hover` is calculated in css

Reply:
I didn't see it happening with the toggle button...

Reply:
On master it definitly does, the wait time is 750ms. But now that I look at master again I need to fix the hover mode since on master it floats over the window (unlike in this PR)

Reply:
> On master it definitly does, the wait time is 750ms. But now that I look at master again I need to fix the hover mode since on master it floats over the window (unlike in this PR)

Yes, unpinned sidebar shouldn't move whole page layout but fly over it like in master.